src/Setup/Template.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Civi\Setup;
+
+class Template {
+
+  protected $filetype;
+
+  protected $smarty;
+  protected $beautifier;
+
+  public function __construct($srcPath, $fileType) {
+    $this->filetype = $filetype;
+
+    $this->smarty = \Civi\Setup\SmartyUtil::createSmarty($srcPath);
+
+    $this->assign('generated', "DO NOT EDIT.  Generated by Installer");
+
+    if ($this->filetype === 'php') {
+      require_once implode(DIRECTORY_SEPARATOR, [$srcPath, 'packages', 'PHP', 'Beautifier.php']);
+      // create an instance
+      $this->beautifier = new PHP_Beautifier();
+      $this->beautifier->addFilter('ArrayNested');
+      // add one or more filters
+      $this->beautifier->setIndentChar(' ');
+      $this->beautifier->setIndentNumber(2);
+      $this->beautifier->setNewLine("\n");
+    }
+  }
+
+  /**
+   * Run template generator.
+   *
+   * @param string $infile
+   *   Filename of the template, without a path.
+   * @param string $outpath
+   *   Full path to the desired output file.
+   */
+  public function getContent($infile, $outpath) {
+    $contents = $this->smarty->fetch($infile);
+
+    if ($this->filetype === 'php') {
+      $this->beautifier->setInputString($contents);
+      $this->beautifier->process();
+      $contents = $this->beautifier->get();
+      // The beautifier isn't as beautiful as one would hope. Here's some extra string fudging.
+      $replacements = [
+        ') ,' => '),',
+        "\n  }\n}\n" => "\n  }\n\n}\n",
+        '=> true,' => '=> TRUE,',
+        '=> false,' => '=> FALSE,',
+      ];
+      $contents = str_replace(array_keys($replacements), array_values($replacements), $contents);
+      $contents = preg_replace('#(\s*)\\/\\*\\*#', "\n\$1/**", $contents);
+      // Convert old array syntax to new square brackets
+      $contents = CRM_Core_CodeGen_Util_ArraySyntaxConverter::convert($contents);
+    }
+
+    return $contents;
+  }
+
+  /**
+   * @param array $inputs
+   *   Template filenames.
+   * @param string $outpath
+   *   Full path to the desired output file.
+   */
+  public function getConcatContent($inputs, $tempOutputFilePath) {
```

Comment:
Maybe I'm not following something, but... can we just remove all the `$tempOutputFilePath` and `file_*_contents()` stuff?

Basically:

```php
$buffer = '';
foreach ($inputs as $infile) {
  $buffer .= $this->smarty->fetch($infile) . "\n";
}
return $buffer;
```